lib/browser/init.js

```
@@ -103,14 +104,15 @@ require('./guest-window-manager')
 // Now we try to load app's package.json.
 let packagePath = null
 let packageJson = null
-const searchPaths = ['app', 'app.asar', 'default_app.asar']
-for (packagePath of searchPaths) {
+for (packagePath of asar.getSearchPath()) {
   try {
     packagePath = path.join(process.resourcesPath, packagePath)
     packageJson = require(path.join(packagePath, 'package.json'))
     break
   } catch (error) {
-    continue
+    if (error.code && error.code.startsWith('EDAMAGED')) {
```

We must not swallow exception is file is damaged.